### DISCUSSION TITLE
How do I use IDOM to preventDefault?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#356] [Question] [answered]
Does IDOM override href's click events to intercept URL routing, or will I need to manually implement this for SPA purposes?

If so, is it a generic selection across all href elements, or only on href components directly rendered by IDOM?

**Comment by rmorshea** [comment, marked as answer]
Instead of just assigning a plain function to an event like `onClick` you can add a decorator to the event handler that encodes extra information like `prevent_default` or `stop_propagation`.

```python
@idom.component
def MyComponent():
    @idom.event(prevent_default=True)
    def handle_click(event):
        ...
    return idom.html.href({"onClick": handle_click}, ...)
```
